RHDEVDOCS-3658- RN Reformat (#41059)
openshift/openshift-docs: modules/cluster-logging-release-notes-5.2.z.adoc

[id="openshift-logging-5-2-6-bug-fixes"]
== Bug fixes
* Before this update, the release did not include a filter change which caused fluentd to crash. With this update, the missing filter has been corrected. (link:https://issues.redhat.com/browse/LOG-2104[LOG-2104])

* This update changes the log4j dependency to 2.17.1 to resolve link:https://access.redhat.com/security/cve/CVE-2021-44832[CVE-2021-44832].(link:https://issues.redhat.com/browse/LOG-2101[LOG-2101])

[id="openshift-logging-5-2-6-CVEs"]
== CVEs
.Click to expand CVEs
[%collapsible]
====
* link:https://access.redhat.com/security/cve/CVE-2021-27292[CVE-2021-27292]
** link:https://bugzilla.redhat.com/show_bug.cgi?id=1940613[BZ-1940613]
* link:https://access.redhat.com/security/cve/CVE-2021-44832[CVE-2021-44832]
** link:https://bugzilla.redhat.com/show_bug.cgi?id=2035951[BZ-2035951]
====

[id="cluster-logging-release-notes-5-2-5"]
= OpenShift Logging 5.2.5

This release includes link:https://access.redhat.com/errata/RHSA-2022:0043[RHSA-2022:0043 OpenShift Logging Bug Fix Release 5.2.5]

[id="openshift-logging-5-2-5-bug-fixes"]
== Bug fixes

> * Before this update, Elasticsearch rejected logs from the eventrouter due to a parsing error. This update changes the data model to resolve the parsing error. However, as a result, existing indices will cause an warning/error within Kibana. The field ("kubernetes.event.metadata.resourceVersion") will cause an error until the removal or reindexing of existing indices.

Reviewer: ```suggestion
* Before this update, Elasticsearch rejected logs from the Event Router due to a parsing error. This update changes the data model to resolve the parsing error. However, as a result, existing indices will cause a warning or error within Kibana. The field ("kubernetes.event.metadata.resourceVersion") will cause an error until the removal or reindexing of existing indices.
```
or "`eventrouter` component"

> 

Reviewer: ```suggestion
+
```

> If this field is not in use in Kibana, then you can ignore the error message. For those with a delete policy, that policy will eventually remove old indices. When that occurs, the message will no longer generate. Those without a delete policy in place will need to manually reindex in order to remove the message.(link:https://issues.redhat.com/browse/LOG-2087[LOG-2087])

Reviewer: ```suggestion
If the `kubernetes.event.metadata.resourceVersion` field is not in use in Kibana, you can ignore the error messages. If you have a delete policy, it eventually removes old indices, which stops the error messages. If you don't have a delete policy, you can manually delete an old index to stop the error messages.(link:https://issues.redhat.com/browse/LOG-2087[LOG-2087])
```
 - It sounds like, if the `kubernetes.event.metadata.resourceVersion` field is in use, must the user take action? Or is it optional? Maybe make this clearer.

> * Before this update, Elasticsearch rejected logs from the eventrouter due to a parsing error. This update changes the data model to resolve the parsing error. However, as a result, existing indices will cause an warning/error within Kibana. The field ("kubernetes.event.metadata.resourceVersion") will cause an error until the removal or reindexing of existing indices.
> 
> If this field is not in use in Kibana, then you can ignore the error message. For those with a delete policy, that policy will eventually remove old indices. When that occurs, the message will no longer generate. Those without a delete policy in place will need to manually reindex in order to remove the message.(link:https://issues.redhat.com/browse/LOG-2087[LOG-2087])

Reviewer: ```suggestion
* Before this update, Elasticsearch rejected logs from the Event Router due to a parsing error. This update changes the data model to resolve the parsing error. However, as a result, previous indices might cause warnings or errors within Kibana. The `kubernetes.event.metadata.resourceVersion` field causes errors until existing indices are removed or reindexed. If this field is not used in Kibana, you can ignore the error messages. If you have a retention policy that deletes old indices, the policy eventually removes the old indices and stops the error messages. Otherwise, manually reindex to stop the error messages. link:https://issues.redhat.com/browse/LOG-2087[LOG-2087])
```

[id="openshift-logging-5-2-5-CVEs"]
== CVEs
.Click to expand CVEs
[%collapsible]
====
* link:https://access.redhat.com/security/cve/CVE-2021-3712[CVE-2021-3712]
* link:https://access.redhat.com/security/cve/CVE-2021-20321[CVE-2021-20321]
* link:https://access.redhat.com/security/cve/CVE-2021-42574[CVE-2021-42574]
* link:https://access.redhat.com/security/cve/CVE-2021-45105[CVE-2021-45105]
====

[id="cluster-logging-release-notes-5-2-4"]
= OpenShift Logging 5.2.4

This release includes link:https://access.redhat.com/errata/RHSA-2021:5127[RHSA-2021:5127 OpenShift Logging Bug Fix Release 5.2.4]

[id="openshift-logging-5-2-4-bug-fixes"]
== Bug fixes

> * Before this update records shipped via syslog would serialize a ruby hash encoding key/value pairs to contain a '=>' character, as well as replace tabs with "#11".  This update serializes the message correctly as proper JSON. (link:https://issues.redhat.com/browse/LOG-1775[LOG-1775])

Reviewer: Duplicate of LOG-1494? If not, use the same wording.

> * Before this update, the Elasticsearch Prometheus exporter plugin was compiling index-level metrics using a high-cost query which had a negative performance impact on the processing Elasticsearch node. With this update, this query has been replaced with a low-impact query that will not impact performance. (link:https://issues.redhat.com/browse/LOG-1970[LOG-1970])

Reviewer: ```suggestion
* Before this update, the Elasticsearch Prometheus exporter plugin compiled index-level metrics using a high-cost query that impacted the Elasticsearch node performance. This update implements a lower-cost query that improves performance. (link:https://issues.redhat.com/browse/LOG-1970[LOG-1970])
```
```suggestion
* Before this update, the Elasticsearch Prometheus exporter compiled index-level metrics by used a high-cost query plugin to . This plugin negatively impacted the performance of the processing Elasticsearch node. This update replaces the previous plugin with one that uses a low-impact query that provides better performance. (link:https://issues.redhat.com/browse/LOG-1970[LOG-1970])
```

> * Before this update, it was possible for Elasticsearch to reject messages because LogForwarding was configured with multiple outputs. Configuration for one of the outputs modified message content to be a single message. This update duplicates the message so output specific processing does not affect other outputs. (link:https://issues.redhat.com/browse/LOG-1824[LOG-1824])

Reviewer: ```suggestion
* Before this update, Elasticsearch sometimes rejected messages when Log Forwarding was configured with multiple outputs. This happened because configuring one of the outputs modified message content to be a single message. This update fixes the issue: it duplicates the message so that output-specific processing does not affect other outputs. (link:https://issues.redhat.com/browse/LOG-1824[LOG-1824])
```

Reviewer: I'm not sure if I understand the issue. This suggestion reflects my best guess or interpretation of what it means. If you apply this suggestion, def ask a QE or SME to review it.

* link:https://www.redhat.com/security/data/cve/CVE-2021-36084.html[CVE-2021-36084]
* link:https://www.redhat.com/security/data/cve/CVE-2021-36085.html[CVE-2021-36085]
* link:https://www.redhat.com/security/data/cve/CVE-2021-36086.html[CVE-2021-36086]
* link:https://www.redhat.com/security/data/cve/CVE-2021-36087.html[CVE-2021-36087]
* link:https://www.redhat.com/security/data/cve/CVE-2021-37136.html[CVE-2021-37136]
* link:https://www.redhat.com/security/data/cve/CVE-2021-37137.html[CVE-2021-37137]
* link:https://www.redhat.com/security/data/cve/CVE-2021-42574.html[CVE-2021-42574]
* link:https://www.redhat.com/security/data/cve/CVE-2021-43267.html[CVE-2021-43267]
* link:https://www.redhat.com/security/data/cve/CVE-2021-43527.html[CVE-2021-43527]
* link:https://www.redhat.com/security/data/cve/CVE-2021-44228.html[CVE-2021-44228]
* link:https://www.redhat.com/security/data/cve/CVE-2021-45046.html[CVE-2021-45046]
====

[id="cluster-logging-release-notes-5-2-3"]
= OpenShift Logging 5.2.3

This release includes link:https://access.redhat.com/errata/RHSA-2021:4032[RHSA-2021:4032 OpenShift Logging Bug Fix Release 5.2.3]

[id="openshift-logging-5-2-3-bug-fixes"]
== Bug fixes

> * Before this update, few alerts were found to not include a namespace label and this was not compliance with the OpenShift Monitoring Team's guidelines for writing alerting rules in OpenShift. With this update, all the alerts in Elasticsearch Operator includes a namespace label and follows all the guidelines for writing alerting rules in OpenShift.(link:https://issues.redhat.com/browse/LOG-1857[LOG-1857])

Reviewer: ```suggestion
* Before this update, some alerts did not include a namespace label. This omission doesn't comply with the OpenShift Monitoring Team's guidelines for writing alerting rules in OpenShift. With this update, all the alerts in Elasticsearch Operator include a namespace label and follow all the guidelines for writing alerting rules in OpenShift.(link:https://issues.redhat.com/browse/LOG-1857[LOG-1857])
```

> * Before this update, a regression introduced in a prior release intentionally disabled JSON message parsing. With this update, a log entry's "level" value is set based on: a parsed JSON message that has a "level" field or by applying a regex against the message field to extract a match. (link:https://issues.redhat.com/browse/LOG-1759[LOG-1759])

Reviewer: ```suggestion
* Before this update, a regression introduced in a prior release intentionally disabled JSON message parsing. This update sets a log entry's "level" value based on a "level" field in a parsed JSON message or by using regex to extract a match from a message field. (link:https://issues.redhat.com/browse/LOG-1759[LOG-1759])
```

* link:https://access.redhat.com/security/cve/CVE-2021-33574[CVE-2021-33574]
* link:https://access.redhat.com/security/cve/CVE-2021-35942[CVE-2021-35942]
* link:https://access.redhat.com/security/cve/CVE-2021-36084[CVE-2021-36084]
* link:https://access.redhat.com/security/cve/CVE-2021-36085[CVE-2021-36085]
* link:https://access.redhat.com/security/cve/CVE-2021-36086[CVE-2021-36086]
* link:https://access.redhat.com/security/cve/CVE-2021-36087[CVE-2021-36087]
====

[id="cluster-logging-release-notes-5-2-2"]
= OpenShift Logging 5.2.2

This release includes link:https://access.redhat.com/errata/RHBA-2021:3747[RHBA-2021:3747 OpenShift Logging Bug Fix Release 5.2.2]

[id="openshift-logging-5-2-2-bug-fixes"]
== Bug fixes

* Before this update, the `ClusterLogging` custom resource (CR) applied the value of the `totalLimitSize` field to the Fluentd `total_limit_size` field, even if the required buffer space was not available. With this update, the CR applies the lesser of the two `totalLimitSize` or 'default' values to the Fluentd `total_limit_size` field, resolving the issue.(link:https://issues.redhat.com/browse/LOG-1738[LOG-1738])

> * Before this update,  a regression introduced in a prior release configuration caused the collector to flush its buffered messages before shutdown, creating a delay the termination and restart of collector Pods. With this update, fluentd no longer flushes buffers at shutdown, resolving the issue.(link:https://issues.redhat.com/browse/LOG-1739[LOG-1739])

Reviewer: ```suggestion
* Before this update, a regression introduced in a prior release configuration caused the collector to flush its buffered messages before shutdown, creating a delay the termination and restart of collector Pods. With this update, fluentd no longer flushes buffers at shutdown, resolving the issue.(link:https://issues.redhat.com/browse/LOG-1739[LOG-1739])
```

[id="openshift-logging-5-2-2-CVEs"]
== CVEs
.Click to expand CVEs
[%collapsible]
====
* link:https://www.redhat.com/security/data/cve/CVE-2020-25648.html[CVE-2020-25648]
* link:https://www.redhat.com/security/data/cve/CVE-2021-22922.html[CVE-2021-22922]
* link:https://www.redhat.com/security/data/cve/CVE-2021-22923.html[CVE-2021-22923]
* link:https://www.redhat.com/security/data/cve/CVE-2021-22924.html[CVE-2021-22924]
* link:https://www.redhat.com/security/data/cve/CVE-2021-36222.html[CVE-2021-36222]
* link:https://www.redhat.com/security/data/cve/CVE-2021-37576.html[CVE-2021-37576]
* link:https://www.redhat.com/security/data/cve/CVE-2021-37750.html[CVE-2021-37750]
* link:https://www.redhat.com/security/data/cve/CVE-2021-38201.html[CVE-2021-38201]
====

[id="cluster-logging-release-notes-5-2-1"]
= OpenShift Logging 5.2.1

This release includes link:https://access.redhat.com/errata/RHBA-2021:3550[RHBA-2021:3550 OpenShift Logging Bug Fix Release 5.2.1]

[id="openshift-logging-5-2-1-bug-fixes"]
== Bug fixes

> * Before this update, due to an issue in the release pipeline scripts, the value of the `olm.skipRange` field remained unchanged at `5.2.0` and was not updated when the z-stream number, `0`, increased. This update fixes the pipeline scripts to update the value of this field when the release numbers change. (link:https://issues.redhat.com/browse/LOG-1743[LOG-1743])

Reviewer: Is the term, "z-stream," Red Hat jargon? If so consider:
```suggestion
* Before this update, due to an issue in the release pipeline scripts, the value of the `olm.skipRange` field remained unchanged at `5.2.0` instead of reflecting the current release number. This update fixes the pipeline scripts to update the value of this field when the release numbers change. (link:https://issues.redhat.com/browse/LOG-1743[LOG-1743])
```
